### PR TITLE
Add `qml.pow` constructor

### PR DESCRIPTION
**Pull request by albi3ro**
This PR finally gets around to adding a top-level `qml.pow` constructor function.  This constructor can be more easily linked than the `__pow__` method and it allows the `lazy=False` keyword option.

```pycon
>>> qml.pow(qml.PauliX(0), 0.5, lazy=False)
SX(wires=[0])
```

### CODE REVIEW

File: pennylane/ops/op_math/pow.py [hunk 1]
```diff
@@ -37,6 +37,59 @@
...
+def pow(base, z=1, lazy=True, do_queue=True, id=None):
+    """Raise an Operator to a power.
+
+    Args:
+        base (~.operation.Operator): the operator to be raised to a power
+        z=1 (float): the exponent
+
+    Keyword Args:
+        lazy=True (bool): In lazy mode, all operations are wrapped in a ``Pow`` class
+            and handled later. If ``lazy=False``, operation-specific simplifications are first attempted.
```
Comment (AlbertMitjans) [inline pennylane/ops/op_math/pow.py:49]: Missing kwargs `do_queue` and `id`

File: pennylane/ops/op_math/pow.py [hunk 2]
```diff
@@ -37,6 +37,59 @@
...
+    if lazy:
+        return Pow(base, z, do_queue=do_queue, id=id)
+    try:
+        pow_ops = base.pow(z)
+    except PowUndefinedError:
+        return Pow(base, z)
+
+    num_ops = len(pow_ops)
+    if num_ops == 0:
+        # needs to be identity (not prod of identities) so device knows to skip
```
Comment (AlbertMitjans) [inline pennylane/ops/op_math/pow.py:79]: Then we should change all the product of identities used in the simplify pipeline...

File: pennylane/ops/op_math/pow.py [hunk 3]
```diff
@@ -37,6 +37,59 @@
...
+
+    Lazy behavior can also be accessed via ``op ** z``.
+
+    """
+    if lazy:
+        return Pow(base, z, do_queue=do_queue, id=id)
+    try:
+        pow_ops = base.pow(z)
+    except PowUndefinedError:
+        return Pow(base, z)
```
Comment (AlbertMitjans) [inline pennylane/ops/op_math/pow.py:75]:
```suggestion
        return Pow(base, z, do_queue=do_queue, id=id)
```

**Review by AlbertMitjans — COMMENTED** [review]
Good job!!

File: pennylane/operation.py [hunk 4]
```diff
@@ -1163,7 +1163,7 @@ def __neg__(self):
     def __pow__(self, other):
         r"""The power operation of an Operator object."""
         if isinstance(other, numbers.Number):
-            return qml.ops.Pow(base=self, z=other)  # pylint: disable=no-member
+            return qml.pow(self, z=other)  # pylint: disable=no-member
```
Comment (AlbertMitjans) [inline pennylane/operation.py:1166]: `test_prod.py` and `test_sprod.py` also use `qml.ops.Pow`, would be nice to change it to `qml.pow`.

Comment (AlbertMitjans) [reply pennylane/operation.py:1166]: Also, we can now remove the `# pylint: disable=no-member` comment.

File: pennylane/ops/op_math/pow.py (hunk 2 above)
Comment (AlbertMitjans) [reply pennylane/ops/op_math/pow.py:83]: Yep, but I don't think we can make it for this release 😢

**Review by AlbertMitjans — APPROVED** [review]
Amazing!!

**Review by timmysilv — APPROVED** [review]
out of scope for this PR but: why do `pow` implementations return lists of Operators? It seems to me that they all return 1-length lists (at least, until now)

**Comment by albi3ro** [comment]
@timmysilv I created it to be the "decomposition of a power operator". decompositions return lists. This also allowed me to easily represent something that decomposed to an identity as an empty list.  We may look into changing this in the future, now that we have `Prod`.

**Pull request merged**